docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+
+![SDK Data Path](./images/SpanDataPath.png)
+
+The specification states: exporter must support two functions: Export and Shutdown.
+
+
+### `Export(span of recordables)`
+
+Exports a batch of telemetry data. Protocol exporters that will implement this function are typically expected to serialize and transmit the data to the destination.
+
+Export() will never be called concurrently for the same exporter instance. Export() can be called again only after the current call returns.
+
+Export() must not block indefinitely, there must be a reasonable upper limit after which the call must time out with an error result (`Failure`).  `exporterTimeoutMillis` - how long the export can run before it is cancelled. The default value is `30000`.
+
+The specification states: Any retry logic that is required by the exporter is the responsibility of the exporter. The default SDK SHOULD NOT implement retry logic, as the required logic is likely to depend heavily on the specific protocol and backend the spans are being sent to.
+
+### `Shutdown()`
+
+Shuts down the exporter. Called when SDK is shut down. This is an opportunity for exporter to do any cleanup required.
+
+`Shutdown` should be called only once for each `Exporter` instance. After the call to `Shutdown` subsequent calls to `Export` are not allowed and should return a `Failure` result.
+
+`Shutdown` should not block indefinitely (e.g. if it attempts to flush the data and the destination is unavailable). Language library authors can decide if they want to make the shutdown timeout configurable.
+
+In the StdoutExporter there is no cleanup to be done, so there is no need to use the timeout within the `Shutdown` function as it will never be blocking. 
+
+```
+class StdoutSpanExporter final : public sdktrace::SpanExporter
+{
+
+private: 
+    bool isShutdown = false;
+    
+public:
+    /* 
+        This function should never be called concurrently.
+    */ 
+    sdktrace::ExportResult Export(
+        const nostd::span<std::unique_ptr<sdktrace::Recordable>> &spans) noexcept
+    {
+    
+        Time startTime = time.now();
+    
+        if(isShutdown)
+        {
+            return sdktrace::ExportResult::kFailure;
+        }
+    
+        for (auto &recordable : spans)
+        {
+            auto span = std::unique_ptr<sdktrace::SpanData>(
+                static_cast<sdktrace::SpanData *>(recordable.release()));
+        
+            if (span != nullptr)
+            {
+                char trace_id[32]       = {0};
+                char span_id[16]        = {0};
+                char parent_span_id[16] = {0};
+        
+                span->GetTraceId().ToLowerBase16(trace_id);
+                span->GetSpanId().ToLowerBase16(span_id);
+                span->GetParentSpanId().ToLowerBase16(parent_span_id);
+        
+                std::cout << "{"
+                        << "\n  name          : " << span->GetName()
```

Comment:
The indentation seems weird, could we make it more readable?

Reply:
Yeah, my bad it got messed up in translation to markdown.